src/pubsub/mod_pubsub.erl

```diff
@@ -1391,18 +1391,23 @@ iq_pubsub_set_create(Host, Node, From,
 
 iq_pubsub_set_publish(Host, Node, From, #{server_host := ServerHost, access := Access,
                                           action_el := ActionEl, query_el := QueryEl}) ->
-    case xml:remove_cdata(ActionEl#xmlel.children) of
-        [#xmlel{name = <<"item">>, attrs = ItemAttrs, children = Payload}] ->
-            ItemId = xml:get_attr_s(<<"id">>, ItemAttrs),
-            PublishOptions = exml_query:path(QueryEl,
-                                             [{element, <<"publish-options">>},
-                                              {element, <<"x">>}]),
-            publish_item(Host, ServerHost, Node, From, ItemId,
-                         Payload, Access, PublishOptions);
-        [] ->
-            {error, extended_error(mongoose_xmpp_errors:bad_request(), <<"item-required">>)};
+    case Node of
```

Comment:
that would be nice to have a separate `iq_pubsub_set_publish/4` function clause matching <<"">> as Node argument instead of this extra `case`